src/Microsoft.Dnx.Runtime/DependencyManagement/LibraryManager.cs

```diff
@@ -78,8 +101,9 @@ private void EnsureInitialized()
                 if (!_initialized)
                 {
                     _initialized = true;
-                    _graph = _libraryInfoThunk().ToDictionary(ld => ld.Name,
-                                                              StringComparer.Ordinal);
+                    var libraries = _librariesThunk();
+                    _graph = libraries.ToDictionary(l => l.Identity.Name, StringComparer.Ordinal);
```

Comment:
Why do we need so many dictionaries? We're going to need to do a serious review of all of our allocations because this looks like overkill.